File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        int stack[10000],top=-1,x=0,y=0,z=0;
        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='(')
            {
                x++;
                stack[++top]=s[i];
            }
             else if(s[i]=='{'){
                y++;
                stack[++top]=s[i];
            }
             else if(s[i]=='[')
             {
                z++;
                stack[++top]=s[i];
            }

            if(top>=0)
            {
              if(s[i]==')' && x>=1 && stack[top]=='(')
              {
                x--;
                top--;
              }
              else if(s[i]=='}'&&y>=1 && stack[top]=='{')
              {
                y--;
                top--;
              }
              else if(s[i]==']'&& z>=1 && stack[top]=='[')
              {
                z--;
                top--;
              }
               else if (s[i] == ']' || s[i] == ')' || s[i] == '}') {  return 0; }
            }else return 0;
        }
        if(top<=-1)
        return true;
        else return 0;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        if len(s)==0:
            return True
        lst=list()
        counter=0
        for i in s:
            if i in '(' or i in '{' or i in '[':
                lst.append(i)
                counter+=1
            elif i in ')' and counter>0:
                if lst[len(lst)-1]=='(':
                    lst.pop()
                    counter-=1
                else:
                    return False
            elif i in '}' and counter>0:
                if lst[len(lst)-1]=='{':
                    lst.pop()
                    counter-=1
                else:
                    return False
            elif i in ']' and counter>0:
                if lst[len(lst)-1]=='[':
                    lst.pop()
                    counter-=1
                else:
                    return False
            else:
                return False

        return True if len(lst)==0 else False